internal/mode/static/manager.go

```diff
@@ -265,22 +265,28 @@ func registerControllers(
 		{
 			objectType: &gatewayv1.GatewayClass{},
 			options: []controller.Option{
+				controller.WithK8sPredicate(k8spredicate.GenerationChangedPredicate{}),
 				controller.WithK8sPredicate(predicate.GatewayClassPredicate{ControllerName: cfg.GatewayCtlrName}),
 			},
 		},
 		{
 			objectType: &gatewayv1.Gateway{},
 			options: func() []controller.Option {
+				options := []controller.Option{
```

Comment:
I want to double-check with the maintainers: Is it correct to add the `GenerationChangedPredicate` option regardless of the value of `GatewayNsName`?

Reply:
I think that's right. We still see other Gateways and keep track of them.

Reply:
If our current behavior is to only watch the Gateway we care about, I think we would only want predicates to apply in that case.

Reply:
The predicates and `NamespacedNameFilter` are applied in different places. Predicates are applied before the controller's `Reconcile` method is called, and the `NamespacedNameFilter` is applied in the `Reconcile` method. So, we can't selectively apply the predicate to a single Gateway without refactoring the code and changing the `NamespacedNameFilter` to a predicate. 

I think what you have here works.  Basically, we first filter out all Gateway upsert events where the generation hasn't changed. Then, in `Reconcile` we apply the `NamespacedNameFilter` and further filter out the Gateways that don't match the `GatewayNsName`.

Reply:
Thanks for the clarification @kate-osborn